### DISCUSSION TITLE
Openbox and LabWC

### DISCUSSION THREAD

**Discussion by AzumaHazuki** [lxqt/lxqt#2445] [General]
To my knowledge, the developer of LabWC is not going to support a good deal of the theming primitives Openbox has. If we intend to use LabWC as the Wayland compositor's don't-call-it-a-default-LXQt-is-WM-agnostic-yew-gaaaaaaize, we have a problem: _LabWC is **not** a drop-in replacement for Openbox._

This means that a user will have a different, and degraded, experiences when starting LXQt exactly the same way bar using Wayland instead of X. I don't imagine porting the missing theming primitives to a fork of LabWC would be _too_ hard for a good coder, but it runs the risk of damaging LabWC itself by becoming the de facto version of the project, not to mention drawing the ire of the original creator...oh, right, and adding a tremendous maintenance burden to LXQt.

What should be done about this? Switching to KWin by default for X and Wayland, forking and adding code to LabWC, or something else entirely?

**Comment by tsujan** [comment]
> If we intend to use LabWC as the Wayland compositor'...

As mentioned on so many occasions,

 * LXQt has no default WM,
 * Distros are free to provide any WMs with their LXQt meta-packages,
 * Users are free to have any DE-agnostic WM they prefer, and
 * Openbox died years ago as a project (although it's quite usable with single-monitor setups).

As for Wayland, currently some LXQt components work with it, some don't. A better Wayland support will be possible only after Qt 6.6.

Thanks to @stefonarch, some practical info concerning the current version of LXQt + Wayland is available here: https://github.com/stefonarch/LXQt-Wayland-files

As another LXQt member who has experimented with Wayland, I've used several LXQt components — except for lxqt-panel and lxqt-runner — under Wayfire, but I had to patch Wayfire, e.g., to use pcmanfm-qt's desktop. Hopefully such patches won't be needed with Qt6.6 + layer-shell-qt6.

**Reply by stefonarch** [reply]
Window manager and wayland compositors theming is regarding only titlebar, borders  and app- and workspace switchers, a relatively small part of the GUI. Of the  2 stacking WM which I consider usable (labwc and wayfire) labwc has the most vast variety of themes and buttons, only gradients are not supported atm.
 You can check if this list here grows by time: https://github.com/labwc/labwc/blob/master/docs/themerc

**Reply by stefonarch** [reply]
>only gradients are not supported atm.

<img width="601" height="572" alt="immagine" src="https://github.com/user-attachments/assets/f3a92fef-8715-49f5-b3d1-ab16fee3a2a3" />

**Comment by AzumaHazuki** [comment]
This is really good to see! I hope LabWC becomes 1:1 feature complete themewise with Openbox someday soon. That will provide a truly seamless X-->Wayland transition.
